src/vm/comthreadpool.cpp

```diff
@@ -751,7 +724,7 @@ FCIMPL1(FC_BOOL_RET, ThreadPoolNative::CorPostQueuedCompletionStatus, LPOVERLAPP
     HELPER_METHOD_FRAME_BEGIN_RET_1(overlapped); // Eventually calls BEGIN_SO_INTOLERANT_CODE_NOTHROW
 
     // OS doesn't signal handle, so do it here
-    overlapped->Internal = 0;
+    lpOverlapped->Internal = 0;
 
     if (ETW_EVENT_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_Context, ThreadPoolIOEnqueue))
         FireEtwThreadPoolIOEnqueue(lpOverlapped, OBJECTREFToObject(overlapped), false, GetClrInstanceId());
```

Comment:
@vancem @brianrob Are you aware of any tooling that uses the object in this event and that can get broken by this change?

Reply:
I am not aware of any tooling that would be affected by moving the overlapped structure from the managed to native heap.  Thanks for checking.

I suspect that in addition to improved throughput on alloc/free, this will also have a positive impact on the heap in cases where PinnableBufferCache wasn't able to do as well as we would have liked.